### PR TITLE
feat(Spinner): Add spinner component

### PR DESCRIPTION
**Pull request by framitdavid**
Created the new spinner component based on Figma sketch. https://www.figma.com/file/vpM9dqqQPHqU6ogfKp5tlr/Felles---Core-Components?node-id=9104%3A49626&t=E0vLjMXEaaBIgtzU-0

### CODE REVIEW

**Review by mimarz — CHANGES_REQUESTED** [review]
Pretty Spinner! Works as advertised 👍 

Added some suggestions for docs, improving DX and allowing more control and expected behaviour from react component.

I see we have some improvements to do in Storybook to make ease of documenting components. I'll some issues regarding this.

Suggest also using [Github keywords ](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for linking it to the issue. Suggest using the resolve keyword.

File: packages/react/src/components/Spinner/Spinner.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    variant: {
+      control: {
+        type: 'select',
+        options: ['default', 'interaction', 'inverted'],
+      },
+    },
+    screenReaderLabel: {
+      type: 'string',
+    },
+  }}
```
[the remark this answers is not in the document]
Comment (framitdavid) [reply packages/react/src/components/Spinner/Spinner.stories.mdx:39]: I agree, was just copying the other stories to make it consistent within the codebase. I think we should consider changing the way we are writing stories. :)

File: packages/react/src/components/Spinner/Spinner.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  screenReaderLabel: string;
+  size?: Size;
+  variant?: Variant;
+};
+export const Spinner = ({
+  screenReaderLabel,
+  size = 'medium',
+  variant = 'default',
+}: SpinnerProps): JSX.Element => {
+  const sizeMap: Record<Size, number> = {
```
[the remark this answers is not in the document]
Comment (framitdavid) [reply packages/react/src/components/Spinner/Spinner.tsx:26]: I agree that those static values can be outside the render function. I prefer to have the Record type to ensure the key is consistent with the type. Record<**Size**, number>, this ensures that the map-object contains only keys that are defined within the type.

File: packages/react/src/components/Spinner/Spinner.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    xSmall: 13,
+    small: 20,
+    medium: 27,
+    large: 40,
+    '1xLarge': 44,
+    '2xLarge': 56,
+    '3xLarge': 79,
+  };
+
+  const variantMap: Record<
```
[the remark this answers is not in the document]
Comment (framitdavid) [reply packages/react/src/components/Spinner/Spinner.tsx:36]: Yeap! :)

File: packages/react/src/components/Spinner/Spinner.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    inverted: {
+      foreground: classes.invertedForeground,
+      background: classes.invertedBackground,
+    },
+  };
+
+  return (
+    <div
+      data-testid='spinner-container'
```
[the remark this answers is not in the document]
Comment (framitdavid) [reply packages/react/src/components/Spinner/Spinner.tsx:56]: Generally, we should avoid using data-testid totally. Using test-id should be a warning of potentially bad accessibility support. We should be able to query almost everything by using getByRole etc. Anyway, for this component, I removed the <div/> and the consumer of this component needs to ensure they stay accessible based on the use case of the spinner.

File: packages/react/src/components/Spinner/Spinner.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      background: classes.interactionBackground,
+    },
+    inverted: {
+      foreground: classes.invertedForeground,
+      background: classes.invertedBackground,
+    },
+  };
+
+  return (
+    <div
```
[the remark this answers is not in the document]
Comment (framitdavid) [reply packages/react/src/components/Spinner/Spinner.tsx:55]: Removed, ref. previous comment. :)

File: packages/react/src/components/Spinner/Spinner.test.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+
+import { Spinner } from './Spinner';
+
```
[the remark this answers is not in the document]
Comment (framitdavid) [reply packages/react/src/components/Spinner/Spinner.test.tsx:5]: I prefer to not test CSS within our unit tests. Based on the conversation we had I do not write more tests for this component. :)

File: packages/react/src/components/Spinner/Spinner.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  | '2xLarge'
+  | '3xLarge';
+
+type Variant = 'default' | 'interaction' | 'inverted';
+
+type SpinnerProps = {
+  screenReaderLabel: string;
+  size?: Size;
+  variant?: Variant;
+};
```
[the remark this answers is not in the document]
Comment (framitdavid) [reply packages/react/src/components/Spinner/Spinner.tsx]: Thanks for the great feedback!

Regarding _"Maybe more of a preference but if you write the string literal directly in SpinnerProps it gets listed out in the IDE intellisense. You decide"_. The reason why I have created types is to reuse the type within the Record to ensure the map object keys stay consistent with the props. :)

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Review by mimarz — CHANGES_REQUESTED** [review]
Seems you are missing and `index.ts` with export of the component and subsequently export in `src/index.ts`. Was this intended?

**Review by mimarz — APPROVED** [review]
thumbs-up-borat

**Pull request merged**